### ISSUE TITLE
Timestamp format does not match documentation

### ISSUE THREAD

**Issue by Bonko** [oauth2-proxy/oauth2-proxy#366]
## Expected Behavior

When logging a timestamp the format should be `19/Mar/2015:17:20:19 -0400` as documented

## Current Behavior

The logged timestamp has the format `2020/01/29 09:54:08`

## Possible Solution

Change the documentation or even better, change the format.
Looks like it is configured here: https://github.com/pusher/oauth2_proxy/blob/d9362d3bb92e610ace7edff59db686783f379a54/pkg/logger/logger.go#L272-L278

## Steps to Reproduce (for bugs)

Check log output

## Context

Our internal log parser expects the documented format to parse the timestamp and fails doing so

## Your Environment

- Version used:
 * 4.0.0

**Comment by JoelSpeed** [comment]
The bump from v3.2.0 to v4.0.0 changes the format, the documentation is therefore incorrect as of v4.0.0

This is obviously in need of a fix though I'm not sure how best to approach this. On the one hand, it would be good to keep the current format so we don't have another breaking change, so perhaps the format could become an option? On the other hand, we have way too many options already so I'm hesitant to suggest adding any more

My suggestions would be that we either update the docs and say, this is what it is now (not helpful for you I realise, sorry), or we add an option to allow people to override it, but set the current option as the default, WDYT @Bonko?

**Comment by JoelSpeed** [comment]
This still needs updating

**Comment by Bonko** [comment]
Hi Joel,

sorry for my late reply.
I already started writing a comment some time ago, but then stopped and forgot about it.

You are right that there are already many configuration options, but this does not have to be a bad thing and others could benefit from it as well.

Regarding changing the "default" back: Another breaking change might be bad, that's right.
I wonder why the format was changed at all. Before it was in the common apache format, which is relatively common.

To sum up:
While I would like this to be in a format that is parseable out of the box by our log parser (or being configurable), I have no hard opinion if this should be implemented or not.

At least the documentation should be correct :)

**Comment by JoelSpeed** [comment]
I think for this one, I'm tempted to just update the documentation and call it there.

Is the current format parsable at all by your log parser?

**Comment by Bonko** [comment]
I don't know if it is worth the effort. We are using (enhanced) grok patterns and the timestamp from `COMBINEDAPACHELOG` does not match, so every entry  is reported at 1.1.1970. However there is still the timestamp when the log arrives at logstash, on which we rely now

**Comment by JoelSpeed** [comment]
Ok, I'm going to suggest not changing the format again and just updating the docs as a solution here

**Comment by JoelSpeed** [comment]
We still need to fix this as far as I remember
